security-command-center/snippets/src/main/java/management/api/CreateEventThreatDetectionCustomModule.java

```diff
@@ -0,0 +1,103 @@
+ * limitations under the License.
+ */
+
+package management.api;
+
+// [START securitycenter_create_event_threat_detection_custom_module]
+import com.google.cloud.securitycentermanagement.v1.CreateEventThreatDetectionCustomModuleRequest;
+import com.google.cloud.securitycentermanagement.v1.EventThreatDetectionCustomModule;
+import com.google.cloud.securitycentermanagement.v1.EventThreatDetectionCustomModule.EnablementState;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import java.io.IOException;
+
+public class CreateEventThreatDetectionCustomModule {
+
+  public static void main(String[] args) throws IOException {
+    // https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules/create
+    // TODO: Developer should replace project_id with a real project ID before running this code
+    String projectId = "project_id";
+
+    String customModuleDisplayName = "custom_module_display_name";
+
+    createEventThreatDetectionCustomModule(projectId, customModuleDisplayName);
+  }
+
+  public static EventThreatDetectionCustomModule createEventThreatDetectionCustomModule(
+      String projectId, String customModuleDisplayName) throws IOException {
+
+    // Initialize client that will be used to send requests. This client only needs
+    // to be created
+    // once, and can be reused for multiple requests.
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+
+      String name =
+          String.format(
+              "projects/%s/locations/global/eventThreatDetectionCustomModules/%s",
+              projectId, "custom_module");
+
+      // define the metadata and other config parameters severity, description,
+      // recommendation below
+      Struct metadataStruct =
+          Struct.newBuilder()
+              .putFields("severity", Value.newBuilder().setStringValue("MEDIUM").build())
+              .putFields(
+                  "description",
+                  Value.newBuilder().setStringValue("add your description here").build())
+              .putFields(
+                  "recommendation",
+                  Value.newBuilder().setStringValue("add your recommendation here").build())
+              .build();
+      Struct configStruct =
+          Struct.newBuilder()
+              .putFields("metadata", Value.newBuilder().setStructValue(metadataStruct).build())
+              .putFields(
+                  "ips",
+                  Value.newBuilder()
+                      .setListValue(
+                          ListValue.newBuilder()
+                              .addValues(Value.newBuilder().setStringValue("0.0.0.0").build())
+                              .build())
+                      .build())
+              .build();
```

Comment:
nit: Do you mean there is no protobuf record for the module's configuration? This long configuration setting looks very confusing and hard to read. Please, confirm with the API/ client library team that there is no other way. If there is no other way, please refactor the code to make it readable. Consider defining the properties in a map and to use protobuf method to convert the map to the protobuf structure.

Reply:
Addressed